### PR TITLE
fix: elastic indices datasource boolean parsing

### PR DESCRIPTION
**Pull request by jdesnoes**
Hello,

A little fix on the elastic indices datasource. When parsing boolean settings the following error could occur:
```
Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: failed to convert setting to bool
│ 
│   with data.elasticstack_elasticsearch_indices.existing_indices["my-indice"],
│   on main.tf line 56, in data "elasticstack_elasticsearch_indices" "existing_indices":
│   56: data "elasticstack_elasticsearch_indices" "existing_indices" {
│ 
│ expected setting to be a bool but got %!t(string=false)
```

@tobio I think a fast review could let it be included in the 0.11.9 version.

Thanks.

### CODE REVIEW

File: internal/elasticsearch/index/indices/models.go [hunk 1]
```diff
@@ -284,6 +284,20 @@ func setSettingsFromAPI(ctx context.Context, model *indexTfModel, apiModel model
...
+				if err != nil {
+					return diag.Diagnostics{
+						diag.NewErrorDiagnostic(
+							"failed to convert setting to bool",
+							fmt.Sprintf("expected setting to be a bool but it was a string. Attempted to parse it but got %s", err.Error()),
+						),
+					}
+				}
+
+				settingsValue = bool(settingBool)
```
Comment (tobio) [inline internal/elasticsearch/index/indices/models.go]:
```suggestion
				settingsValue = settingBool
```

**Review by tobio — COMMENTED** [review]
Thanks @jdesnoes, are you able to add a changelog entry here?

File: internal/elasticsearch/index/indices/models.go (hunk 1 above)
Comment (jdesnoes) [reply internal/elasticsearch/index/indices/models.go]: Thanks a lot @tobio for your fast review.

My bad for the changelog, I forgot about...
I just pushed a new commit with the changelog updated and your code suggestion applied. Thanks.

**Comment by jdesnoes** [comment]
@tobio it looks there is failing test step on 8.8.2 but I don't see anything related to the changes I did with this PR.
I guess everything should be OK if we try to run it again?

**Review by tobio — APPROVED** [review]

**Comment by tobio** [comment]
Thanks for coming back to fix this up!

**Pull request merged**